### ISSUE TITLE
Documentation on how to make custom extractors

### ISSUE THREAD

**Issue by sergiomeneses** [Totodore/socketioxide#374]
**Is your feature request related to a problem? Please describe.**
I want to create a custom extrators for my app.

**Describe the solution you'd like**
Add an example of a extrator that access state.

**Describe alternatives you've considered**
Just docs

**Additional context**
I need to create my custom extrators but i found so far the foloowing issues:

1. When i implement `FromMessageParts` (or any of these handler traits), i can't get the `json_serde::Value` and`bytes::Bytes` Ttpes (you should export those.
2. If i need my custom extrator to access socket state, the only solution at the moment is to create the state from the current message (also for other type handlers) i.e. `State::<T>::from_message_parts(s, v, p, ack_id)` which returns `Result<Self, StateNotFound<T>>`, so i have consum the result and covert to option using `ok()`, and finally i `unwrap()` the option, which is safe in my context but i think we need a better way to do this kind of extractors.

**Comment by Totodore** [comment]
I agree that custom extractor implementations lack examples and docs. 

However I'm curious about what you are trying to do, because if you want to access the axum state you simply have to make a shared state and add it to both axum and socketioxide with [`with_state`](https://docs.rs/socketioxide/latest/socketioxide/struct.SocketIoBuilder.html#method.with_state) and then you can use the State extractor provided by socketioxide.
But maybe I didn't understand you original issue.

**Comment by sergiomeneses** [comment]
I want to access socket state not Axum state, I want to inject my database repositories in handler arguments (just I do with Axum) so I can't skip boilerplate in my socket handlers

**Comment by Totodore** [comment]
For your first point:
I re-exported the socketioxide `Value` type #382 to easily implement extractor traits. It will be available in the next major version. However if you want to manually deserialize it to a specific type you need to use `Data::from_{connect,message,disconnect}_parts`.

For you second point:
I agree that accessing the state is currently complex. The state system deserve an improvement to match the axum state system. I still want to wait before modifying this part.

For your third point:
I don't understand what you want to do? `Data<(Type1, Type2, Type3)>` will deserialize data to a tuple of three types according to the incoming payload.

**Comment by Totodore** [comment]
Hey,
regarding the first issue I still don't know yet how to solve this issue so you still have to use the State extractor manually.

For the second I didn't know that extractors where implemented for tuple types and this is a good idea.
I'll make a PR to add this feature.

**Comment by Totodore** [comment]
Re-opening this issue to add blanket tuple implementation for extractors.
